pkg/report/writer_test.go

```diff
@@ -311,6 +311,294 @@ func TestReportWriter_Template(t *testing.T) {
+		.severity-CRITICAL .severity { background-color: #e40000; }
+		.severity-UNKNOWN .severity { background-color: #747474; }
+		.severity-LOW { background-color: #5fbb3160; }
+		.severity-MEDIUM { background-color: #e9c60060; }
+		.severity-HIGH { background-color: #ff880060; }
+		.severity-CRITICAL { background-color: #e4000060; }
+		.severity-UNKNOWN { background-color: #74747460; }
+		table tr td:first-of-type {
+		font-weight: bold;
+		}
+		.links a,
+		.links[data-more-links=on] a {
+		display: block;
+		}
+		.links[data-more-links=off] a:nth-of-type(1n+5) {
+		display: none;
+		}
+		a.toggle-more-links { cursor: pointer; }
+	</style>
+	<title>{{- escapeXML ( index . 0 ).Target }} - Trivy Report - {{ getCurrentTime }}</title>
+	<script>
+		window.onload = function() {
+		document.querySelectorAll('td.links').forEach(function(linkCell) {
+			var links = [].concat.apply([], linkCell.querySelectorAll('a'));
+			[].sort.apply(links, function(a, b) {
+			return a.href > b.href ? 1 : -1;
+			});
+			links.forEach(function(link, idx) {
+			if (links.length > 3 && 3 === idx) {
+				var toggleLink = document.createElement('a');
+				toggleLink.innerText = "Toggle more links";
+				toggleLink.href = "#toggleMore";
+				toggleLink.setAttribute("class", "toggle-more-links");
+				linkCell.appendChild(toggleLink);
+			}
+			linkCell.appendChild(link);
+			});
+		});
+		document.querySelectorAll('a.toggle-more-links').forEach(function(toggleLink) {
+			toggleLink.onclick = function() {
+			var expanded = toggleLink.parentElement.getAttribute("data-more-links");
+			toggleLink.parentElement.setAttribute("data-more-links", "on" === expanded ? "off" : "on");
+			return false;
+			};
+		});
+		};
+	</script>
+	</head>
+	<body>
+	<h1>{{- escapeXML ( index . 0 ).Target }} - Trivy Report - {{ getCurrentTime }}</h1>
+	<table>
+	{{- range . }}
+		<tr class="group-header"><th colspan="6">{{ escapeXML .Type }}</th></tr>
+		{{- if (eq (len .Vulnerabilities) 0) }}
+		<tr><th colspan="6">No Vulnerabilities found</th></tr>
+		{{- else }}
+		<tr class="sub-header">
+		<th>Package</th>
+		<th>Installed Version</th>
+		<th>Level</th>
+		<th>Vulnerability ID</th>
+		<th>Fixed Version</th>
+		<th>Links</th>
+		</tr>
+		{{- range .Vulnerabilities }}
+		<tr class="severity-{{ escapeXML .Vulnerability.Severity }}">
+		<td class="pkg-name">{{ escapeXML .PkgName }}</td>
+		<td class="pkg-version">{{ escapeXML .InstalledVersion }}</td>
+		<td class="severity">{{ escapeXML .Vulnerability.Severity }}</td>
+		<td>{{ escapeXML .VulnerabilityID }}</td>
+		<td>{{ escapeXML .FixedVersion }}</td>
+		<td class="links" data-more-links="off">
+			{{- range .Vulnerability.References }}
+			<a href={{ escapeXML . | printf "%q" }}>{{ escapeXML . }}</a>
+			{{- end }}
+		</td>
+		</tr>
+		{{- end }}
+		{{- end }}
+	{{- end }}
+	</table>
+{{- else }}
+	</head>
+	<body>
+	<h1>Trivy Returned Empty Report</h1>
+{{- end }}
+	</body>
+</html>`,
+			expected: `<!DOCTYPE html>
```

Comment:
You didn't add any logic in `writer.go`. So, I think this case is not necessary and looks hard to maintain. Could you add the case in integration tests as follows?
https://github.com/aquasecurity/trivy/blob/675e1b4118d206ac9d0b212c3844ff4cd13d2ce6/integration/client_server_test.go#L313-L322

Reply:
Done! But i had to change
https://github.com/aquasecurity/trivy/blob/bcbe45ce459b7101c7d76caf21fbdd2b450c7a4c/integration/client_server_test.go#L530
from  JSONEq to EqualValues.

Reply:
This looks reasonable to me. WDYT @knqyf263?